File: implement/Intervalares/bissection.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

//Function
double f(double x)
{
  return x * x - 4;
}

double f1(double x)
{
  return x * x * x - 10 * sin(x) + 2.8;
}

void bissection(double f(double), double a, double b, int numIt)
{
  double m;
  int counter = 0;

  for (int i = 0; i < numIt; ++i)
  {
    m = (a + b) / 2.0;

    if (f(a) * f(m) >= 0)
      a = m;
    else
      b = m;

    cout << "|a - b| = " << abs(a - b) << endl;
    cout << "a = " << a << endl;
    cout << "b = " << b << endl;
    cout << "m = " << m << endl;

    counter++;
  }
}

int main()
{
  const int OUT_PREC = 5;

  cout << fixed << setprecision(OUT_PREC);

  bissection(f1, 1.5, 4.2, 3);

  return 0;
}

File: implement/Intervalares/bissection.py
import math


def f(x):
	return x**3 -10*math.sin(x) + 5.5




def bissection(f,a,b,numIT):
	i = 0
	if(f(a)*f(b) < 0):
		while(i<numIT):
			x = (a+b)/2
			if f(a) * f(x) < 0:
				b = x
			else:
				a = x
			x = (a+b)/2
			print(i)
			i+=1

			print("A: ",a)
			print("B: ",b)
		print("Root: ", x)
		print(f(x))
			
	else:
		print("No roots in given interval")

bissection(f,1.5,5.6,2)